File: testing/iOS/StyleConverterTest/StyleEngine/spacing/GapConfig.swift
//
//  GapConfig.swift
//  StyleEngine/spacing — Phase 2.
//
//  CSS `gap` splits into RowGap + ColumnGap; the fixture converter always
//  expands shorthand to both longhands. Each side carries a Phase 1
//  LengthValue so the applier can resolve em/percent/viewport the same
//  way as padding + margin.
//

// Foundation for consistency.
import Foundation

struct GapConfig: Equatable {
    // Spacing between rows in a flex column / grid.
    var row: LengthValue = .exact(px: 0)
    // Spacing between columns in a flex row / grid.
    var column: LengthValue = .exact(px: 0)

    // True when either dimension carries a non-zero value. Callers use
    // this to decide whether to override the container's default spacing.
    var hasAny: Bool {
        !isZero(row) || !isZero(column)
    }

    private func isZero(_ v: LengthValue) -> Bool {
        if case .exact(0) = v { return true }
        return false
    }
}

File: testing/Android/app/src/main/java/com/styleconverter/test/style/spacing/GapConfig.kt
package com.styleconverter.test.style.spacing

// GapConfig — container-level row-gap and column-gap. CSS shorthand `gap: N`
// expands to `row-gap: N; column-gap: N` (no standalone Gap property reaches
// us — the converter always emits both longhands). See
// examples/properties/spacing/gap.json for fixture coverage.
//
// IR shapes (DIFFERENT from padding/margin — all tagged):
//   {"type": "length", "px": 10}                 → Exact 10px
//   {"type": "length", "original": {"v":1,"u":"EM"}}  → Relative EM
//   {"type": "percentage", "value": 5}           → Relative PERCENT

import com.styleconverter.test.style.core.types.LengthValue

/**
 * Row / column gap for a flex or grid container. Null = unspecified,
 * renderer falls back to 0 (the CSS initial value).
 */
data class GapConfig(
    val rowGap: LengthValue? = null,
    val columnGap: LengthValue? = null,
) {
    val hasGap: Boolean
        get() = rowGap != null || columnGap != null
}
